server/gateway/services/UserService.js

```diff
@@ -0,0 +1,217 @@
+
+  /**
+   * Creates a user
+   *
+   * @param {object} args - The request arguments passed in from the controller
+   * @param {IncomingMessage} response - The http response object
+   * @param {function} next - The callback used to pass control to the next action/middleware
+   */
+  createUser: async function (args, response, next) {
+
+    var userRequest = args.user.value;
+
+    var request = new Message(
+      PubSubChannels.User.External.Event,
+      constants.pubSub.messageType.crud,
+      constants.pubSub.messageAction.create,
+      userRequest
+    );
+
+    try {
+      let completed =
+        await PubSub.publishAndWaitForResponse(
+          PubSubChannels.User.External.Event,
+          PubSubChannels.User.External.CompletedEvent,
+          {
+            subscriberType: constants.pubSub.recipients.gateway
+          },
+          request);
+
+      response.statusCode = completed.payload.statusCode;
+      response.setHeader('Content-Type', 'application/json');
+      return response.end(JSON.stringify(completed.payload.body));
+    } catch (err) {
+      logging.logAction(
+        logging.logLevels.ERROR,
+        `Failed to subscribe to channel [${PubSubChannels.User.External.CompletedEvent}]`, err);
+      return next(err);
+    }
+  },
+
+  /**
+   * Returns all users
+   *
+   * @param {object} args - The request arguments passed in from the controller
+   * @param {IncomingMessage} response - The http response object
+   * @param {function} next - The callback used to pass control to the next action/middleware
+   */
+  getAllUsers: async function (args, response, next) {
+
+    var request = new Message(
+      PubSubChannels.User.External.Event,
+      constants.pubSub.messageType.crud,
+      constants.pubSub.messageAction.getAll,
+      {}
+    );
+
+    try {
+      let completed =
+        await PubSub.publishAndWaitForResponse(
+          PubSubChannels.User.External.Event,
+          PubSubChannels.User.External.CompletedEvent,
+          {
+            subscriberType: constants.pubSub.recipients.gateway
+          },
+          request);
+
+      response.statusCode = completed.payload.statusCode;
+      response.setHeader('Content-Type', 'application/json');
+      response.setHeader('X-Result-Count', completed.payload.header.resultCount);
+      return response.end(JSON.stringify(completed.payload.body));
+
+    } catch (err) {
+      logging.logAction(
+        logging.logLevels.ERROR,
+        `Failed to subscribe to channel [${PubSubChannels.User.External.CompletedEvent}]`, err);
+      return next(err);
+    }
+  },
+
+  /**
+   * Returns a single user
+   *
+   * @param {object} args - The request arguments passed in from the controller
+   * @param {IncomingMessage} response - The http response object
+   * @param {function} next - The callback used to pass control to the next action/middleware
+   */
+  getSingleUser: async function (args, response, next) {
+
+    var userId = args.id.value;
```

Comment:
Will we be using userId to reference to a user entry? Would it make more sense to use MSISDN?